### PR TITLE
Functions that allow marks must also deal with unknown values

### PR DESCRIPTION
**Pull request by jbardin**
You can't deal with marks in a function without also dealing with unknowns. If `AllowMarked` is true, but `AllowKnown` is not, the caller will short-circuit evaluation with an unknown, without being able to deal with marks from all arguments. The cty package can't guarantee that an unknown value contains all possible marks (some functions could alter marks based on the value), but it definitely helps to catch unexpected marks as early as possible during evaluation.

### CODE REVIEW

File: internal/lang/funcs/funcs_test.go
Comment (DanielMSchmidt) [inline internal/lang/funcs/funcs_test.go:1]: Was this added by accident?

**Review by DanielMSchmidt — APPROVED** [review]
It sounds like this would fix these skipped tests:

- https://github.com/hashicorp/terraform/blob/main/internal/terraform/context_plan_ephemeral_test.go#L275
- https://github.com/hashicorp/terraform/blob/5d41e0142609715dbcd9cb4d504fe9df54cd0e51/internal/terraform/context_plan_ephemeral_test.go#L302

File: internal/lang/funcs/funcs_test.go
Comment (jbardin) [reply internal/lang/funcs/funcs_test.go:1]: This initializes the global logger, so that logs aren't emitted during tests without `-v`.

**Comment by jbardin** [comment]
Thanks, I'll check on those `toBeImplemented` tests here too. It seems there are a few other things that need followup in the `terraform` package anyway.

**Pull request merged**